src/Microsoft.Data.SqlClient/src/Microsoft/Data/Common/DbConnectionStringCommon.cs

```diff
@@ -810,6 +867,65 @@ internal static SqlConnectionAttestationProtocol ConvertToAttestationProtocol(st
             }
         }
 
+        internal static SqlConnectionEncryptionOption ConvertToSqlConnectionEncryptionOption(string keyword, object value)
+        {
+            if (value is null)
+            {
+                return DbConnectionStringDefaults.Encrypt;
+            }
+            if (value is string sValue)
+            {
+                // try again after remove leading & trailing whitespaces.
+                sValue = sValue.Trim();
+                if (TryToConvertToSqlConnectionEncryptionOption(sValue, out SqlConnectionEncryptionOption result))
+                {
+                    return result;
+                }
+                // string values must be valid
+                throw ADP.InvalidConnectionOptionValue(keyword);
+            }
+            else
```

Comment:
I think we can remove the `else` here because the condition above is a guard clause since it either returns the result or throws an exception, and that's one less layer of indention for readability